src/rules/no-system-props.js

```diff
@@ -0,0 +1,138 @@
+const {isPrimerComponent} = require('../utils/isPrimerComponent')
+const {pick} = require('@styled-system/props')
+const {some, last} = require('lodash')
+
+// Components for which we allow all styled system props
+const excludedComponents = new Set([
+  'Box',
+  'Text',
+  'BaseStyles' // BaseStyles will be deprecated eventually
+])
+
+// Components for which we allow a set of prop names
+const excludedComponentProps = new Map([
+  ['AnchoredOverlay', new Set(['width', 'height'])],
+  ['Avatar', new Set(['size'])],
+  ['Dialog', new Set(['width', 'height'])],
+  ['Label', new Set(['variant'])],
```

Comment:
`Flash` also accepts a `variant` prop:
```suggestion
  ['Dialog', new Set(['width', 'height'])],
  ['Flash', new Set(['variant'])],
  ['Label', new Set(['variant'])],
```